megamek/src/megamek/client/ui/swing/boardview/BoardViewTooltipProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing.boardview;
+
+import megamek.common.Coords;
+import megamek.common.annotations.Nullable;
+
+import java.awt.*;
+import java.awt.event.MouseEvent;
+
+/**
+ * This interface is implemented by classes that can be added to a BoardView to provide the tooltip contents
+ * for the BoardView.
+ *
+ * @see BoardView#setTooltipProvider(BoardViewTooltipProvider)
+ */
+public interface BoardViewTooltipProvider {
```

Comment:
It seems like we normally use "I" as the leading character in an Interface name, although I'm not sure if that's by convention or a rule.

Reply:
I wouldnt say that its a rule, rather the opposite. I try not using the I* when I can avoid it. I only use it when the fitting interface name is taken by the present class name and I want to avoid having 140 files changed because of a name change. I know its happened rather often in my PRs tho. Changing the name of Game would be a real multi-hit wonder. There'll be a time when I do that tho :)